src/dc_tools.rs

```diff
@@ -341,19 +339,24 @@ pub unsafe fn dc_utf8_strlen(s: *const libc::c_char) -> size_t {
 
     j
 }
+pub fn dc_truncate_n_str(buf: &str, approx_chars: usize, do_unwrap: bool) -> Cow<str> {
+    let ellipse = if do_unwrap { "..." } else { "[...]" };
 
-pub fn dc_truncate_str(buf: &str, approx_chars: usize) -> Cow<str> {
-    if approx_chars > 0 && buf.len() > approx_chars + ELLIPSE.len() {
+    if approx_chars > 0 && buf.len() > approx_chars + ellipse.len() {
         if let Some(index) = buf[..approx_chars].rfind(|c| c == ' ' || c == '\n') {
-            Cow::Owned(format!("{}{}", &buf[..index + 1], ELLIPSE))
+            Cow::Owned(format!("{}{}", &buf[..index + 1], ellipse))
         } else {
-            Cow::Owned(format!("{}{}", &buf[..approx_chars], ELLIPSE))
+            Cow::Owned(format!("{}{}", &buf[..approx_chars], ellipse))
         }
     } else {
         Cow::Borrowed(buf)
     }
 }
 
+pub fn dc_truncate_str(buf: &str, approx_chars: usize) -> Cow<str> {
```

Comment:
I find the naming difference between these two funcs not obvious.  But I also don't have anything better to suggest.

Reply:
In general we only need one of them, so we should replace all the other ones with this one.